src/pages/iou/IOUModal.js

```diff
@@ -145,6 +145,9 @@ class IOUModal extends Component {
 
     componentDidMount() {
         PersonalDetails.fetchLocalCurrency();
+        if (this.props.iou.selectedCurrencyCode) {
+            return;
+        }
         IOU.setIOUSelectedCurrency(this.props.myPersonalDetails.localCurrencyCode);
```

Comment:
NAB: I think the early return looks a bit strange in the componenDidMount method. I'd opt for 
```js
if(!this.props.iou.selectedCurrencyCode) {
    IOU.setIOUSelectedCurrency(this.props.myPersonalDetails.localCurrencyCode);
}
```

Reply:
Initially, I did that but changed as I saw lint error. But maybe it was because I wrapped whole function body inside it. Anyways, updated it